### PR TITLE
Added support for I2.xlarge ssd

### PR DESCRIPTION
**Pull request by tomerk**

### CODE REVIEW

**Comment by tomerk** [comment]
This currently only adds the first SSD for the rest of the I2 instance types.
There are probably other instance types that need to be updated in the same way.
Also, we need to figure out what to do with the pssh timeout: these larger disks can take long enough to format that pssh times out and the disks don't get attached correctly. For now I've set the timeout to 300 seconds at @shivaram 's suggestion, but this might not be long enough for some machines or too long for others.

File: setup-slave.sh [hunk 1]
```diff
@@ -45,6 +46,18 @@ if [[ $instance_type == r3* ]]; then
   fi
 fi
 
+if [[ $instance_type == i2* ]]; then
```
Comment (shivaram) [inline setup-slave.sh]: Can we move this to the same block as `r3` i.e. have `$instance_type == r3* || $instance_type == i2*`

**Comment by shivaram** [comment]
Thanks @tomerk for the PR. I left a style comment

@nchammas -- Could you comment on the `pssh` timeout ? AFAIK the issue seems to be that formatting the 800GB drives on the i2 machines takes longer than the default `pssh` timeout. Is there a better way to ask `pssh` to wait for the process to exit ?

File: setup.sh [hunk 2]
```diff
@@ -70,6 +70,7 @@ pssh --inline \
     --host "$MASTERS $SLAVES" \
     --user root \
     --extra-args "-t -t $SSH_OPTS" \
+    --timeout 300 \
```
Comment (nchammas) [inline setup.sh]: @tomerk What timeout do you see without this setting? The man page doesn't mention a default value.

It does mention, though, that you can set the timeout to 0 and pssh will wait until the called processes complete. Does that not work here?

Comment (tomerk) [inline setup.sh]: It's a 60 second timeout by default. I did have it set to 0 before but @Shivaram suggested that that might not be optimal

—
Best Wishes,
Tomer Kaftan

Comment (shivaram) [inline setup.sh]: Actually does setting it to 0 match our previous `ssh` loop ? If so, it should be fine to set it to 0.

Comment (nchammas) [inline setup.sh]: It [doesn't look like](https://github.com/mesos/spark-ec2/blob/9f67e96e4763a03931c0b1ab3f26b6faa8f6266e/setup.sh#L30) we had anything in previously to set the timeout apart from a connection timeout (which is preserved with the pssh call). So the SSH timeout would just depend on the OS's default `TCPKeepalive` setting, I believe.

Comment (shivaram) [inline setup.sh]: Ok - The TCPKeepAlive (essentially wait for the process to finish until TCP connections are up)) sounds similar to setting timeout 0 for pssh. So I'd suggest doing that. 
@tomerk It'll be great if you can test if `timeout = 0` works correctly.

Comment (tomerk) [inline setup.sh]: Yeah I previously had timeout = 0 set and it seemed to be working correctly.

**Comment by shivaram** [comment]
Thanks @tomerk - Looks good. @nchammas Any other comments ?

**Comment by nchammas** [comment]
Nope, LGTM.

**Comment by tomerk** [comment]
We should probably also open a ticket somewhere for seeing what other instance types need to be updated.

**Comment by shivaram** [comment]
Good point - Can you open an issue at https://issues.apache.org/jira/browse/SPARK ?

**Pull request merged**